sdk/core/core-lro/src/lroEngine/operation.ts

```diff
@@ -135,8 +134,7 @@ export class GenericPollOperation<TResult, TState extends PollOperationState<TRe
   }
 
   async cancel(): Promise<PollOperation<TState, TResult>> {
-    await this.cancelOp?.(this.state);
-    return this;
+    throw new Error("This method is deprecated because it wasn't implemented");
   }
```

Comment:
It is a change in behavior for sure but is it necessarily a breaking change? Shouldn't we alert those customers that they're not actually canceling operation when they call `cancelOperation`?

Reply:
Yeah, we definitely want to alert our customers, but I think throwing an error is too severe for this situation. It will turn a success path into a failure path with no code changes in consumers.

 I don't think the amount of disruption this error can cause pays off for letting the user know that this is basically a no-op. If we were throwing to fail fast (i.e .cancel would eventually cause an exception anyway) then it should be fine.

Reply:
I think logging a warning should be a good enough alert for a minor version bump.

Reply:
Ok, addressed in [d099a86](https://github.com/Azure/azure-sdk-for-js/pull/22844/commits/d099a869c47792376ce694f4ce4843a22a4c62a4)